azavea/ashlar
# Support Python 3

# Overview

This PR adds support for Python 3. After merging, Ashlar will support Python 3.3-3.6, in addition to Python 2.7.

## Notes

- Most of the changes to the app and tests involve altering statements so that they work across Python versions. In cases where the reason behind the change is unclear, I've left a review comment explaining my reasoning.

## Testing instructions

I took three steps to confirm that this PR added appropriate support:

1. Updated to Python 3 in the [reference implementation](https://github.com/azavea/ashlar-blueprint) and played around to confirm that everything still worked as expected
2. Configured Travis to test all of the appropriate Python versions
3. Added [new tests](https://github.com/azavea/ashlar/pull/83) to make sure that all critical code paths were executed

Since https://github.com/azavea/ashlar/pull/83 (improve test coverage) is a branch off of this PR, it should be considered the canonical test that this branch works. You can view its tests in Travis, or you can follow the testing instructions there.


Closes https://github.com/azavea/ashlar-2018-fellowship/issues/9.
